src/context/wishListContext.js

```diff
@@ -0,0 +1,75 @@
+import { createContext, useContext, useReducer } from "react";
+import { useFetchApi } from "../hooks/useFetchApi";
+import { useAuth } from "./";
+import { initialSharedState } from "../reducers/constants";
+import { sharedReducer } from "../reducers/reducerFunctions/sharedReducer";
+import axios from "axios";
+
+const WishListContext = createContext();
+
+const WishListProvider = ({ children }) => {
+  const [wishlistState, wishlistDispatch] = useReducer(
+    sharedReducer,
+    initialSharedState
+  );
+  const apiData = {
+    api: "/api/user/wishlist",
+    property: "wishList",
+  };
+  const { userAuthState } = useAuth();
+  const { authToken } = userAuthState;
+
+  const config = {
+    headers: {
+      authorization: authToken,
+    },
+  };
+  useFetchApi(apiData, config);
+
+  const addToWishlist = async (product) => {
+    try {
+      const res = await axios.post(
+        "/api/user/wishlist",
+        {
+          product,
+        },
+        config
+      );
+      wishlistDispatch({ type: "SET_DATA", payload: res.data.wishlist });
```

Comment:
suggestion - you can add check for success status code & then dispatch

Reply:
sure will check this